chains/orchestrator-relays/runtime/starlight/src/lib.rs

```diff
@@ -0,0 +1,3813 @@
+    SnowbridgeTanssi(ChannelId),
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl From<u32> for AggregateMessageOrigin {
+    fn from(n: u32) -> Self {
+        // Some dummy for the benchmarks.
+        Self::Ump(UmpQueueId::Para(n.into()))
+    }
+}
+
+pub struct GetAggregateMessageOrigin;
+
+impl Convert<ChannelId, AggregateMessageOrigin> for GetAggregateMessageOrigin {
+    fn convert(channel_id: ChannelId) -> AggregateMessageOrigin {
+        AggregateMessageOrigin::Snowbridge(channel_id)
+    }
+}
+
+impl Convert<UmpQueueId, AggregateMessageOrigin> for GetAggregateMessageOrigin {
+    fn convert(queue_id: UmpQueueId) -> AggregateMessageOrigin {
+        AggregateMessageOrigin::Ump(queue_id)
+    }
+}
+
+pub struct GetAggregateMessageOriginTanssi;
+
+impl Convert<ChannelId, AggregateMessageOrigin> for GetAggregateMessageOriginTanssi {
+    fn convert(channel_id: ChannelId) -> AggregateMessageOrigin {
+        AggregateMessageOrigin::SnowbridgeTanssi(channel_id)
+    }
+}
+
+/// This is used by [parachains_inclusion::Pallet::on_queue_changed]
+pub struct GetParaFromAggregateMessageOrigin;
+
+impl Convert<AggregateMessageOrigin, ParaId> for GetParaFromAggregateMessageOrigin {
+    fn convert(x: AggregateMessageOrigin) -> ParaId {
+        match x {
+            AggregateMessageOrigin::Ump(UmpQueueId::Para(para_id)) => para_id,
+            AggregateMessageOrigin::Snowbridge(channel_id)
+            | AggregateMessageOrigin::SnowbridgeTanssi(channel_id) => {
+                // Read para id from EthereumSystem::channels storage map
+                match EthereumSystem::channels(channel_id) {
+                    Some(x) => x.para_id,
+                    None => {
+                        // This should be unreachable, but return para id 0 if channel does not exist
+                        log::warn!(
+                            "Got snowbridge message from channel that does not exist: {:?}",
+                            channel_id
+                        );
+                        ParaId::from(0)
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// The relay register and deregister calls should no longer be necessary
+/// Everything is handled by the containerRegistrar
+pub struct IsRelayRegister;
+impl Contains<RuntimeCall> for IsRelayRegister {
+    fn contains(c: &RuntimeCall) -> bool {
+        matches!(
+            c,
+            RuntimeCall::Registrar(paras_registrar::Call::register { .. })
+        ) || matches!(
+            c,
+            RuntimeCall::Registrar(paras_registrar::Call::deregister { .. })
+        )
+    }
+}
+
+/// Starlight shouold not permit parathread registration for now
+/// TODO: remove once they are enabled
+pub struct IsParathreadRegistrar;
+impl Contains<RuntimeCall> for IsParathreadRegistrar {
+    fn contains(c: &RuntimeCall) -> bool {
+        matches!(
+            c,
+            RuntimeCall::ContainerRegistrar(pallet_registrar::Call::register_parathread { .. })
```

Comment:
same, we should avoid all calls to containerRegistrar

Reply:
fixed